src/Kit/System/Watchdog/_0test/watchdog.cpp

```diff
@@ -0,0 +1,374 @@
+ *
+ * Redistributions of the source code must retain the above copyright notice.
+ *----------------------------------------------------------------------------*/
+/** @file */
+
+#include "Kit/System/Watchdog/Supervisor.h"
+#include "Kit/System/Watchdog/WatchedThread.h"
+#include "Kit/System/Watchdog/WatchedEventThread.h"
+#include "Kit/System/_testsupport/ShutdownUnitTesting.h"
+#include "catch2/catch_test_macros.hpp"
+#include "Kit/System/ElapsedTime.h"
+#include "Kit/System/Trace.h"
+#include "Kit/System/Api.h"
+#include "Kit/System/TimerManager.h"
+
+#define SECT_                  "_0test"
+
+#define TEST_TIMEOUT_SHORT_MS  100
+#define TEST_TIMEOUT_MEDIUM_MS 1000
+#define TEST_TIMEOUT_LONG_MS   2000
+#define TEST_SLEEP_SHORT_MS    50
+#define TEST_SLEEP_MEDIUM_MS   100
+#define TEST_SLEEP_LONG_MS     200
+
+///
+using namespace Kit::System;
+
+
+////////////////////////////////////////////////////////////////////////////////
+namespace {
+
+/// Test WatchedEventThread that simulates health check failure
+class FailingHealthCheckThread : public WatchedEventThread
+{
+public:
+    ///
+    bool m_shouldFailHealthCheck;
+    ///
+    int m_healthCheckCallCount;
+
+public:
+    ///
+    FailingHealthCheckThread( uint32_t wdogTimeoutMs, uint32_t healthCheckIntervalMs, bool isSupervisor = false )
+        : WatchedEventThread( wdogTimeoutMs, healthCheckIntervalMs, isSupervisor )
+        , m_shouldFailHealthCheck( false )
+        , m_healthCheckCallCount( 0 )
+    {
+    }
+
+protected:
+    /// Override to provide controllable health check failure
+    bool performHealthCheck() noexcept override
+    {
+        m_healthCheckCallCount++;
+        KIT_SYSTEM_TRACE_MSG( SECT_, "Health check called %d times, should fail: %s", 
+                             m_healthCheckCallCount, m_shouldFailHealthCheck ? "true" : "false" );
+        return !m_shouldFailHealthCheck;
+    }
+};
+
+}  // end anonymous namespace
+
+
+////////////////////////////////////////////////////////////////////////////////
+TEST_CASE( "watchdog" )
+{
+    KIT_SYSTEM_TRACE_FUNC( SECT_ );
+
+    SECTION( "hal functions" )
+    {
+        KIT_SYSTEM_TRACE_MSG( SECT_, "Testing HAL wrapper functions" );
+
+        // Test enabling watchdog
+        bool enabled = enableWdog();
+        REQUIRE( enabled == true );
+        KIT_SYSTEM_TRACE_MSG( SECT_, "Watchdog enabled successfully" );
+
+        // Test kicking watchdog (should not crash)
+        kickWdog();
+        KIT_SYSTEM_TRACE_MSG( SECT_, "Watchdog kicked successfully" );
+
+        // Test tripping watchdog (in stub mode, should not actually reset)
+        tripWdog();
+        KIT_SYSTEM_TRACE_MSG( SECT_, "Watchdog tripped successfully (stub mode)" );
+    }
+
+    SECTION( "supervisor basic operations" )
+    {
+        KIT_SYSTEM_TRACE_MSG( SECT_, "Testing Supervisor class basic operations" );
+
+        // Test enabling watchdog
+        bool enabled = Supervisor::enableWdog();
+        REQUIRE( enabled == true );
```

Comment:
You need to do more than check the return code, i.e. verify that the HAL interface was called
```suggestion
        bool enabled = Supervisor::enableWdog();
        REQUIRE( enabled == true );
        REQUIRE(  watchdogEnabled_ == true );  // See previous comment about moving the mocked HAL methods and variables into this file
```